bin/git-forgit

```diff
@@ -71,6 +71,55 @@ hash emojify &>/dev/null && _forgit_emojify='|emojify'
 # extract the first git sha occurring in the input and strip trailing newline
 _forgit_extract_sha="grep -Eo '[a-f0-9]+' | head -1 | tr -d '[:space:]'"
 
+# parse a space separated string into an array
+# arrays parsed with this function are global
+_forgit_parse_array() {
+    ${IFS+"false"} && unset old_IFS || old_IFS="$IFS"
+    # read the value of the second argument
+    # into an array that has the name of the first argument
+    IFS=" " read -r -a "$1" <<< "$2"
+    ${old_IFS+"false"} && unset IFS || IFS="$old_IFS"
+}
+
+# parse environment variables into global arrays
+# this allows using them properly quoted without word splitting or glob expansion
+_forgit_log_git_opts=()
+_forgit_parse_array _forgit_log_git_opts "$FORGIT_LOG_GIT_OPTS"
+_forgit_add_git_opts=()
+_forgit_parse_array _forgit_add_git_opts "$FORGIT_ADD_GIT_OPTS"
+_forgit_diff_git_opts=()
+_forgit_parse_array _forgit_diff_git_opts "$FORGIT_DIFF_GIT_OPTS"
+_forgit_reset_head_git_opts=()
+_forgit_parse_array _forgit_reset_head_git_opts "$FORGIT_RESET_HEAD_GIT_OPTS"
+_forgit_rebase_git_opts=()
+_forgit_parse_array _forgit_rebase_git_opts "$FORGIT_REBASE_GIT_OPTS"
+_forgit_fixup_git_opts=()
+_forgit_parse_array _forgit_fixup_git_opts "$FORGIT_FIXUP_GIT_OPTS"
+_forgit_blame_git_opts=()
+_forgit_parse_array _forgit_blame_git_opts "$FORGIT_BLAME_GIT_OPTS"
+_forgit_revert_commit_git_opts=()
+_forgit_parse_array _forgit_revert_commit_git_opts "$FORGIT_REVERT_COMMIT_GIT_OPTS"
+_forgit_branch_delete_git_opts=()
+_forgit_parse_array _forgit_branch_delete_git_opts "$FORGIT_BRANCH_DELETE_GIT_OPTS"
+_forgit_checkout_commit_git_opts=()
+_forgit_parse_array _forgit_checkout_commit_git_opts "$FORGIT_CHECKOUT_COMMIT_GIT_OPTS"
+_forgit_checkout_tag_git_opts=()
+_forgit_parse_array _forgit_checkout_tag_git_opts "$FORGIT_CHECKOUT_TAG_GIT_OPTS"
+_forgit_checkout_branch_git_opts=()
+_forgit_parse_array _forgit_checkout_branch_git_opts "$FORGIT_CHECKOUT_BRANCH_GIT_OPTS"
+_forgit_checkout_file_git_opts=()
+_forgit_parse_array _forgit_checkout_file_git_opts "$FORGIT_CHECKOUT_FILE_GIT_OPTS"
+_forgit_cherry_pick_git_opts=()
+_forgit_parse_array _forgit_cherry_pick_git_opts "$FORGIT_CHERRY_PICK_GIT_OPTS"
+_forgit_clean_git_opts=()
+_forgit_parse_array _forgit_clean_git_opts "$FORGIT_CLEAN_GIT_OPTS"
+_forgit_stash_push_git_opts=()
+_forgit_parse_array _forgit_stash_push_git_opts "$FORGIT_STASH_PUSH_GIT_OPTS"
+_forgit_stash_show_git_opts=()
+_forgit_parse_array _forgit_stash_show_git_opts "$FORGIT_STASH_SHOW_GIT_OPTS"
+_forgit_checkout_branch_branch_git_opts=()
+_forgit_parse_array _forgit_checkout_branch_branch_git_opts "$FORGIT_CHECKOUT_BRANCH_BRANCH_GIT_OPTS"
```

Comment:
Is there any performance impact of something like this? Does this mean every time we call a `forgit` command, we are populating the arrays for all commands, not just reading the variable for the one we are calling?

Reply:
@cjappl That is indeed a good point!
@sandr01d Can we move the parsing of each variable into the corresponding individual function?

See, splitting out individual chunks of a large PR always creates focus and brings up new valuable input! :-)

Reply:
Good point, I've pushed a commit with the change. Will have to move things around a bit when I rebase #326 onto this, as the arrays will mostly move into the `_forgit_git_*` functions. There are two arrays we'll be parsing twice: `_forgit_diff_git_opts` and `_forgit_blame_git_opts` because both are used in the main function (e.g. `_forgit_blame`) and the preview function. We do not make a sacrifice on performance here though, because the preview functions will be executed from a fzf subshell and the arrays need to be reparsed therefore anyway.